include/kickcat/protocol.h

```diff
@@ -29,13 +47,22 @@ namespace kickcat
     constexpr int32_t  ETH_MAX_SIZE = sizeof(EthernetHeader) + ETH_MTU_SIZE + ETH_FCS_SIZE;
     constexpr int32_t  ETH_MIN_SIZE = 60; // Ethernet disallow sending less than 64 bytes (60 + FCS)
 
+    using EthernetFrame = std::array<uint8_t, ETH_MAX_SIZE>; // Definition of an Ethernet frame (maximal size)
 
     // EtherCAT description
-    constexpr uint16_t ETH_ETHERCAT_TYPE = __builtin_bswap16(0x88A4); //TODO: implement constexpr htons
+    constexpr uint16_t ETH_ETHERCAT_TYPE = hton<uint16_t>(0x88A4);
     constexpr int32_t ETHERCAT_WKC_SIZE = 2;
     constexpr int32_t MAX_ETHERCAT_DATAGRAMS = 15;       // max EtherCAT datagrams per Ethernet frame
     constexpr uint16_t MAX_ETHERCAT_PAYLOAD_SIZE = 1486; // max EtherCAT payload size possible
 
+    // EtherCAT subprotocol
+    enum EthercatType : int16_t
+    {
+        ETHERCAT = 0x1,             // EtherCAT Data Link Protocol Data Unit (the real time protocol)
+        NETWORK_VARIABLES = 0x4,
```

Comment:
what are network variables ?

Reply:
Another network protocol that seems to be designed to let multiple masters to communicates. I add this entry to have all the possibles types in the code.